caravel/tasks.py

```diff
@@ -0,0 +1,160 @@
+# https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Explain
+def get_tables(sql):
+    # TODO(bkyryliuk): implement
+    return []
+
+
+def create_table_as(sql, table_name):
+    """Reformats the query into the create table as query.
+
+    :param sql: string, sql query that will be executed
+    :param table_name: string, will contain the results of the query execution
+    :return: string, create table as query
+    """
+    # TODO(bkyryliuk): split into atomic sql queries
+    # TODO(bkyryliuk): drop table if allowed, check the namespace and
+    #                  the permissions.
+
+    sql_statements = sqlparse.split(sql)
+    updated_sql = ""
+    for sql_statement in sql_statements:
+        if not sql_statement:
+            continue
+        if is_query_select(sql):
+            sql_statement = "CREATE TABLE IF NOT EXISTS %s AS %s" % (
+                table_name, sql_statement)
+        updated_sql += sql_statement + "\n"
+    return updated_sql
+
+
+def get_session():
+    """Creates new SQLAlchemy scoped_session."""
+    engine = create_engine(
+        app.config.get('SQLALCHEMY_DATABASE_URI'), convert_unicode=True)
+    return scoped_session(sessionmaker(
+        autocommit=False, autoflush=False, bind=engine))
+
+
+@celery_app.task
+def get_sql_results(database_id, sql, user_id):
+    """Executes the sql query returns the results.
+
+    :param database_id: integer
+    :param sql: string, query that will be executed
+    :param user_id: integer
+    :return: dataframe, query result
+    """
+    # Create a separate session, reusing the db.session leads to the
+    # concurrency issues.
+    query_session = get_session()
+    try:
+        db_to_query = query_session.query(models.Database).filter_by(
+            id=database_id).first()
+    except Exception as e:
+        return {'error': utils.error_msg_from_exception(e)}
+
+    if not db_to_query:
+        return {
+            'error': "Database with id {0} is missing.".format(database_id)}
+
+    query = init_query(database_id, sql, user_id, int(time.time() * 1000))
+    query_session.add(query)
+    query_session.commit()
+
+    query_result, success = get_sql_results_internal(sql, db_to_query)
+
+    query.end_time = int(time.time() * 1000)
+    if success:
+        query.query_status = models.QueryStatus.FINISHED
+        # TODO(bkyryliuk): fill in query.tmp_table_name
+    else:
+        query.query_status = models.QueryStatus.FAILED
+
+    query_session.commit()
+    # TODO(bkyryliuk): return the tmp table  / query_id
+    return query_result
+
+
+# TODO(bkyryliuk): merge the changes made in the carapal first
+#                   before merging this PR.
+def get_sql_results_internal(sql, db_to_query):
+    """Get the SQL query resulst from the give session and db connection.
+
+    :param sql: string, query that will be executed
+    :param db_to_query: models.Database to query
+    :return: (dataframe, boolean), results and the status
+    """
+    try:
```

Comment:
agree, do u have suggestions how to figure out the types of exceptions I should catch here?
DB implementation provide their own exceptions, for example _mysql_exceptions.ProgrammingError and the sqlalchemy api doesn't have a list of them.

I've come to catching all of those statements via trying out the tests on the travis on multiple DB implementations.